server/handler/check_run.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/google/go-github/github"
+	"github.com/palantir/go-githubapp/githubapp"
+	"github.com/pkg/errors"
+
+	"github.com/palantir/policy-bot/pull"
+)
+
+type CheckRun struct {
+	Base
+}
+
+func (h *CheckRun) Handles() []string { return []string{"check_run"} }
+
+func (h *CheckRun) Handle(ctx context.Context, eventType, deliveryID string, payload []byte) error {
+	var event github.CheckRunEvent
+	if err := json.Unmarshal(payload, &event); err != nil {
+		return errors.Wrap(err, "failed to parse check_run event payload")
+	}
+
+	if event.GetAction() != "completed" || event.GetCheckRun().GetConclusion() != "success" {
```

Comment:
Wouldn't we want to post a status check regardless of the conclusion of the check run?

Reply:
At the moment, we care about check runs only for the `has_successful_status` predicate. Evaluating on success saves some computation (the status handler does the same thing) without hurting correctness.